WordPress/gutenberg
# Author, Author Bio, Author Name: Add a fallback for Author Archive Template

Fixes #55410
Similar to #40648

Props to @them-es for reporting the issue with [this comment](https://github.com/WordPress/gutenberg/pull/55352#issuecomment-1763832410).

## What?

This PR fixes an issue where the block is not displayed in the Author Archive template or a warning error occurs when the author has no posts in the following three blocks.

- Avatar
- Author Name
- Author Biography

## Why?

These three blocks assume that a post ID context (`$block->context['postId']`) exists. But my understanding is that if that author hasn't posted any articles yet, the post ID context doesn't exist, so the block won't show up or you'll get an error.

Since the Author Archive page is accessible, I think they should be rendered correctly even in cases where they don't have any posts.

## How?

Use `author` query parameter as ID when no post context exists.

## Testing Instructions

- First, create a user with an avatar. For example, if you register as a user using the email address below, the avatar will be displayed.
  - `test1@example.com`
  - `test2@example.com`
  - `test3@example.com`
- Enter the user's biography (Also, please note the author ID).
- Create the Author Archive template in the Site Editor.
- Enter the following code in the code editor and save it.
  ```
  
  
  
  ```
- Access `http://localhost:8888/?author={user_id}`.
- Although this author does not have any posts, their avatar, name, and biography should all be displayed correctly.
